include/taskolib/SequenceManager.h

```diff
@@ -57,51 +60,127 @@ namespace task {
 class SequenceManager
 {
 public:
-    using PathList = std::vector<std::filesystem::path>;
+    /// A struct to represent a sequence on the disk.
+    struct SequenceOnDisk
+    {
+        std::filesystem::path path; ///< The path to the sequence
+        std::string label; ///< The human-readable label of the sequence
+        UniqueId unique_id; ///< The unique ID of the sequence
+    };
 
     /**
-     * Creates a new instance to manage and control of sequences that are serialized in
-     * the underlying file system.
+     * Create a SequenceManager to manage sequences that are stored in a given directory.
      *
-     * \param path set root path to the sequence folders.
+     * \param path  the root folder that contains individual folders for each sequence.
      *
-     * \exception throws Error exception if path is empty.
+     * \exception Error is thrown if the path name is empty.
      */
-    explicit SequenceManager(std::filesystem::path path)
-    {
-        if (path.empty())
-            throw Error("Root sequences path must not be empty.");
-        path_ = path;
-    }
+    explicit SequenceManager(std::filesystem::path path);
 
     /**
-     * Returns the root path of the serialized sequences.
+     * Create an empty sequence on disk.
+     *
+     * \returns a new sequence with the specified label and a random ID that is unique
+     *          among the sequences inside this folder.
      *
-     * \return root path of the serialized sequences.
+     * \exception Error is thrown if the sequence folder cannot be created.
+     */
+    Sequence create_sequence(gul14::string_view label = "") const;
```

Comment:
In Sequence. h Finni mentioned that the overload of  explicit Sequence(const UniqueId& unique_id); is not necessary because you could call Sequence name with default value "".
Why you choose now to do it here without function overload?

Reply:
I think we at least need two overloads: One with an explicit UID, one without it (that generates a random UID). Giving the label a default is just a matter of convenience.

Reply:
Superseded in new version of the MR.